x-pack/legacy/plugins/canvas/canvas_plugin_src/renderers/time_filter/components/datetime_quick_list/datetime_quick_list.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode, FunctionComponent } from 'react';
+import PropTypes from 'prop-types';
+import { EuiButton, EuiButtonEmpty } from '@elastic/eui';
+import 'react-datetime/css/react-datetime.css';
+
+interface Props {
+  /** Initial start date string */
+  from: string;
+  /** Initial end date string */
+  to: string;
+
+  /** Function invoked when a date range is clicked */
+  onSelect: (from: string, to: string) => void;
+
+  /** Nodes to display under the date range buttons */
+  children?: ReactNode;
+}
+
+const quickRanges = [
```

Comment:
Consider adding these as an export with `TimeUnit` below in some common file.  We'll need to i18n both at some point.

Reply:
I'm a little hesitant to making that move of the `quickRanges` here to a common place since there's another `quickRanges` specific to the `PrettyDuration` component as well. I can refactor them to perhaps share the same `quickRanges` array so that we only need to update the string for i18n in one place if you think that's worth it in this PR

Reply:
I'd track a follow-up.  We have a lot of instances where we have these kinds of ranges, (e.g. refresh, auto-play, etc).

Reply:
Do we want them all to share the same quick ranges? I imagine refresh/autoplay would share the same set of quick range time intervals, but I think time filter should probably have a different set of date ranges from the ones used for refresh/autoplay.

Side note: While looking through the available Kibana advanced settings, I noticed there was a `timepicker:quickRanges` setting which we may want to consider using instead of the static `quickRanges` array used in the `time_filter` component. I've added it to my list of Kibana advanced settings we'd want to integrate into Canvas (https://github.com/elastic/kibana/issues/37436).

Reply:
Tracking that here: https://github.com/elastic/kibana/issues/41055